__main__.py

```diff
@@ -1,28 +1,16 @@
 from simulator.modules import Clients, LoadBalancer, ServerWithGCI
-from utils import get_config, get_logger, generate_results, flag
-import simpy, os, time
+from log import get_logger, log_server_data, log_latency, log_percentiles
+from config import get_config
+import simpy, os, time, math
 
 def create_directory():
     if not os.path.isdir("logs"):
         os.mkdir("logs")
 
-def log_server_data(logger, server):
-    logger.info("Heap level: %.5f%%" % server.heap.level)
-    logger.info("Remaining requests in queue: %i" % len(server.queue.items))
-    logger.info("GCI executions: %i" % server.gci.times_performed)
-    logger.info("GC executions: %i" % server.gc.times_performed)
-    logger.info("GC execution time sum: %.3f seconds" % server.gc.gc_exec_time_sum)
-
-def log_latence(logger, requests):
-    logger.info("Processed requests: %i" % len(requests))
-    logger.info("Latence of the first request: %.3f" % requests[0]._latence_time)
-    logger.info("Latence of the last request: %.3f" % requests[-1]._latence_time)
-    sumx = sum(request._latence_time for request in requests)
-    media = sumx / len(requests)
-    logger.info("Latence media of the requests: %.3f" % media)
+def percentile(P, N):
```

Comment:
def percentile(l, p):
   """Returns the value within the list l representing pth percentile."""
   s = sorted(list(l))
   pos = min(math.ceil((P / 100) * N), len(s))
   return s[pos]

To calculate the percentile using the nearest rank, the list needs to be sorted. There is no comment about that and it makes the function very fragile to be used.

Reply:
Thanks for comment this! Consider fixed.